### ISSUE TITLE
E741: Inconsistent behavior for lambdas

### ISSUE THREAD

**Issue by knkski** [PyCQA/pycodestyle#959]
It seems like E741 shouldn't apply to variable names inside of comprehensions, due to the limited scope of such variables. If that's the desired behavior anyways, this behavior seems inconsistent to me:

```python
# flagged as lint
foo = [l for l in lines if l]

# not flagged as lint
foo = list(filter(lambda l: l, lines))
```

**Comment by asottile** [comment]
I believe it's just a mistake that this isn't handled for lambdas

the rule is correct for the list comprehension due to the ability to confuse:


```python
foo = [l for l in lines if l]
foo = [1 for l in lines if 1]
```

**Comment by pktiuk** [comment]
I have this issue, but in my case it is clearly inconsistent:

```python
COMP_OPERATIONS = {
    "==": lambda l, r: l == r,   # This is falgged
    ">=": lambda l, r: l >= r,   # This is falgged
    "<=": lambda l, r: l <= r,   # This is falgged
    ">": lambda l, r: l > r,  # This is not flagged
    "<": lambda l, r: l < r,  # This is not flagged
}
```

**Comment by asottile** [comment]
@pktiuk yeah it's not enabled for lambdas at all at the moment -- it's detecting `l >= r` as an assignment (?)
